### PR TITLE
Add k8s walkthrough for connection pools

### PR DESCRIPTION
**Pull request by fawadkhaliq**
**Issue** https://github.com/aws/aws-app-mesh-controller-for-k8s/issues/347

**Description of changes**
- Added a k8s walkthrough for connection pools/circuit breakers for virtual gateway and virtual node
- Enabled for preview only. This will be changed when we move out of preview

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: walkthroughs/howto-k8s-connection-pools/README.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+
+## Prerequisites
+This feature is currently only available in [App Mesh preview](https://docs.aws.amazon.com/app-mesh/latest/userguide/preview.html) and will work with App Mesh controller [here](https://github.com/aws/eks-charts/tree/preview/stable/appmesh-controller). App Mesh preview is only provided the `us-west-2` region.
```
Comment (achevuru) [inline walkthroughs/howto-k8s-connection-pools/README.md:10]: nit: provided the -> provided in the

**Review by achevuru — APPROVED** [review]

File: walkthroughs/howto-k8s-connection-pools/README.md (hunk 1 above)
Comment (fawadkhaliq) [reply walkthroughs/howto-k8s-connection-pools/README.md]: Thanks, fixed!

File: walkthroughs/howto-k8s-connection-pools/v1beta2/manifest.yaml.template [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    matchLabels:
+      app: ingress-gw
+  template:
+    metadata:
+      labels:
+        app: ingress-gw
+    spec:
+      containers:
+        - name: envoy
+          image: controller-override
```
Comment (marceloboeira) [inline walkthroughs/howto-k8s-connection-pools/v1beta2/manifest.yaml.template:274]: @fawadkhaliq how does this work? where does this image come from?

Comment (marceloboeira) [reply walkthroughs/howto-k8s-connection-pools/v1beta2/manifest.yaml.template:274]: okay, found it https://github.com/aws/aws-app-mesh-controller-for-k8s/pull/262/files#diff-67827a4affa2ad0d56ada1256f5cb2138356b1b5f5ff47afab7ca15cf3d28444R19

Comment (marceloboeira) [reply walkthroughs/howto-k8s-connection-pools/v1beta2/manifest.yaml.template:274]: that fails for me with an error in the  `spec.containers[1].ports[0].containerPort:` even tho I have a single container only...

Comment (marceloboeira) [reply walkthroughs/howto-k8s-connection-pools/v1beta2/manifest.yaml.template:274]: it seems this only happens when x-ray is enabled for the cluster...

**Pull request merged**